pkg/apis/pipeline/v1alpha1/git_resource_test.go

```diff
@@ -140,37 +182,107 @@ func Test_GitResource_Replacements(t *testing.T) {
 func Test_GitResource_GetDownloadTaskModifier(t *testing.T) {
 	names.TestingSeed()
 
-	r := &v1alpha1.GitResource{
-		Name:     "git-resource",
-		Type:     v1alpha1.PipelineResourceTypeGit,
-		URL:      "git@github.com:test/test.git",
-		Revision: "master",
-		GitImage: "override-with-git:latest",
-	}
-
-	ts := v1alpha1.TaskSpec{}
-	modifier, err := r.GetInputTaskModifier(&ts, "/test/test")
-	if err != nil {
-		t.Fatalf("Unexpected error getting GetDownloadTaskModifier: %s", err)
-	}
-
-	want := []v1alpha1.Step{{Container: corev1.Container{
-		Name:    "git-source-git-resource-9l9zj",
-		Image:   "override-with-git:latest",
-		Command: []string{"/ko-app/git-init"},
-		Args: []string{
-			"-url",
-			"git@github.com:test/test.git",
-			"-revision",
-			"master",
-			"-path",
-			"/test/test",
+	for _, tc := range []struct {
+		desc        string
+		gitResource *v1alpha1.GitResource
+		want        corev1.Container
+	}{
```

Comment:
nit: If you collapse this `{` and the next one, you can save a couple horizontal lines, and a vertical tabwidth:

```
for _, tc := range []struct {
  // struct fields
} {{
  // case 1
}, {
  // case 2
}} {
  t.Run(...)
}
```